### PR TITLE
[feature](hive_write) add hive_staging_dir catalog properties and change default value under the writing target table.

### PR DESCRIPTION
**Pull request by kaka11chen**
### What problem does this PR solve?

Problem Summary:

### Release note

[feature] (hive_write) add hive_staging_dir catalog properties and change default value under the writing target table.

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [x] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by kaka11chen** [comment]
run buildall

**Comment by hello-stephen** [comment]
# FE Regression Coverage Report
Increment line coverage `  0.00% (0/102)` :tada:
[Increment coverage report](http://coverage.selectdb-in.cc/coverage/60018_094e5816a786fef63b65691a4327e0a069c8467e_merge_fe/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/60018_094e5816a786fef63b65691a4327e0a069c8467e_merge_fe/report/index.html)

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HMSTransaction.java [hunk 1]
```diff
@@ -1620,6 +1635,117 @@ public void shutdownExecutorService() {
         }
     }
 
+    private static boolean isSubDirectory(String parent, String child) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HMSTransaction.java]: Add unit test for these static methods.
This check is very important, we need to cover every corner case, otherwise, we may delete data file accidentally

File: fe/fe-core/src/main/java/org/apache/doris/planner/HiveTableSink.java [hunk 2]
```diff
@@ -168,7 +169,14 @@ public void bindDataSink(Optional<InsertCommandContext> insertCtx)
 
     private String createTempPath(String location) {
         String user = ConnectContext.get().getCurrentUserIdentity().getUser();
-        return LocationPath.getTempWritePath(location, "/tmp/.doris_staging/" + user);
+        String defaultStagingBaseDir = ".doris_staging";
+        String stagingBaseDir = targetTable.getCatalog().getCatalogProperty()
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/HiveTableSink.java:173]: Can user specify this stage dir arbitrarily? Is it safe?
What if user specify dir like `/etc` or `/`?

Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/planner/HiveTableSink.java:173]: i think we should only support `/tmp/.staging` or relative dir?

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HMSTransaction.java (hunk 1 above)
Comment (kaka11chen) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HMSTransaction.java]: ok

File: fe/fe-core/src/main/java/org/apache/doris/planner/HiveTableSink.java (hunk 2 above)
Comment (kaka11chen) [reply fe/fe-core/src/main/java/org/apache/doris/planner/HiveTableSink.java:173]: Other systems also use this facility.

**Comment by hello-stephen** [comment]
# FE Regression Coverage Report
Increment line coverage `  45.28% (48/106)` :tada:
[Increment coverage report](http://coverage.selectdb-in.cc/coverage/60018_ab8c962d7bad7af1f8468bda6421868f324ade68_merge_fe/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/60018_ab8c962d7bad7af1f8468bda6421868f324ade68_merge_fe/report/index.html)

**Review by morningman — APPROVED** [review]

**Review by zhangstar333 — APPROVED** [review]
